src/main/java/Duke.java

```diff
@@ -1,10 +1,254 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private final static String LOAD_PATH = "save/tasks.txt";
+
+    private static void addTask(ArrayList<Task> tasks, Task task) {
+        tasks.add(task);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task);
+        System.out.printf("Now you have %d tasks in the list\n", tasks.size());
+    }
+
+    private static void doMark(ArrayList<Task> tasks, String[] wordList) throws DukeException {
+        if (wordList.length != 2) {
+            throw new DukeException("☹ OOPS!!! Wrong number of arguments for mark");
+        }
+        try {
+            int index = Integer.parseInt(wordList[1]) - 1;
+            tasks.get(index).setDone(true);
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(tasks.get(index));
+
+        } catch (Exception exception) {
+            System.out.println(exception);
+        }
+    }
+
+    private static void doUnmark(ArrayList<Task> tasks, String[] wordList) throws DukeException {
```

Comment:
I can see that you have named a lot of your functions like `doSave`, or `doMark`. It might not be very intuitive for the reader to understand what exactly the function is trying to perform so consider naming it something like `saveFile` or `markTask`/`mark`, this applies to the other functions you have named in a similar manner. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain)